src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -366,8 +374,8 @@
       <PackageVersionsPropsContents>
         <![CDATA[
 <Project>
-  <Import Project="$(PreviouslySourceBuiltPackageVersionPropsPath)" />
-  <Import Project="$(CurrentSourceBuiltPackageVersionPropsPath)" />
+  <Import Condition="Exists('$(PreviouslySourceBuiltPackageVersionPropsPath)')" Project="$(PreviouslySourceBuiltPackageVersionPropsPath)" />
+  <Import Condition="Exists('$(CurrentSourceBuiltPackageVersionPropsPath)')" Project="$(CurrentSourceBuiltPackageVersionPropsPath)" />
```

Comment:
These conditions could result in those files silently not getting imported in the common path (that happened). Can we condition the WriteLinesToFile task below instead and the global property that gets passed to the repo build?

Reply:
I'd like to localize the dependencies on this condition all right here. There are other parts of the build that require the package props file to exist. So I changed it to still write to the file, but it will be an empty project with no imports.

Fixed in https://github.com/dotnet/sdk/pull/48327/commits/51fe34256b12965c46bad1a7cde7789057ae9807